apps/dashr-circos/app.R

```diff
@@ -0,0 +1,283 @@
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(jsonlite)
+library(dashBio)
+library(GEOquery)
+
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+  
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+  
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+app <- Dash$new()
+
+
+
+source("assets/CircosFunctions.R")
+
+# Load up the preloaded dataset into R structures.
+
+json_file <- "assets/circos_graph_data.json"
+
+circos_dataframe <- jsonlite::fromJSON("assets/circos_graph_data.json")
+
+
+circos_graph_data <- read_json(path = json_file)
+
+
+applayout <- layout()
```

Comment:
This is problematic because `layout` is both a function in the `graphics` package, as well as a function in the `plotly` package (unfortunately). Sorry, this happens so frequently that I can't help but mention it, you'd be surprised how many people name things `layout` 🙂 

As a general rule, I try to avoid namespace collisions for R functions and other objects. If there's an exported object with the same name I want to use, I'll figure out what I can call it so that I don't unintentionally overload anything.

I'd rename your layout function in the `CircosFunctions.R` script, but you'll either need to add a `library(dashTable)` line or change all references to `dashDataTable` to `dashTable::dashDataTable` since it's not imported into the `NAMESPACE` of the `dash` package automatically. That will make this error go away:

```
 Error in dashDataTable(id = "data-table", columns = lapply(colnames(df),  : 
  could not find function "dashDataTable" 
```